source/crud/tests/v2/aggregate-merge.yml

```diff
@@ -0,0 +1,136 @@
+data:
+    - { _id: 1, x: 11 }
+    - { _id: 2, x: 22 }
+    - { _id: 3, x: 33 }
+
+minServerVersion: '4.1.12'
+
+collection_name: &collection_name 'test_aggregate_merge'
+
+tests:
+  -
+    description: "Aggregate with $merge"
+    operations:
+      -
+        object: collection
+        name: aggregate
+        arguments: &arguments
+          pipeline: &pipeline
+            - $sort: { x : 1 }
+            - $match: { _id: { $gt: 1 } }
+            - $merge: { into: &output_collection "other_test_collection" }
+    expectations:
+      -
+        command_started_event:
+          command:
+            aggregate: *collection_name
+            pipeline: *pipeline
+    outcome: &outcome
+      collection:
+        name: *output_collection
+        data:
+          - { _id: 2, x: 22 }
+          - { _id: 3, x: 33 }
+  -
+    description: "Aggregate with $merge and batch size of 0"
+    operations:
+      -
+        object: collection
+        name: aggregate
+        arguments:
+          <<: *arguments
+          batchSize: 0
+    expectations:
+      -
+        command_started_event:
+          command:
+            aggregate: *collection_name
+            pipeline: *pipeline
+            # TODO: check that cursor.batchSize does not exist once
+            # https://jira.mongodb.org/browse/SPEC-1215 has been resolved.
```

Comment:
It's not clear to me if the rules for comparing APM commands allows us to assert nonexistent fields of embedded documents. See https://github.com/mongodb/specifications/pull/560#pullrequestreview-246810295 for some earlier discussion on this. In any case, I think it's reasonable to wait on [SPEC-1215](https://jira.mongodb.org/browse/SPEC-1215) to sort this out. This comment is derived from some existing comments elsewhere in CRUD spec test files.

Reply:
batchSize is an option specified inside "cursor" so I think you can do:
```
            aggregate: *collection_name
            pipeline: *pipeline
            cursor: {}
```

Reply:
Ah, that makes complete sense. There's no reason for any other field to exist within `cursor` (based on the current command API), so it _must_ be an empty document.